src/pad/src/ICeWall.cpp

```diff
@@ -0,0 +1,1070 @@
+                                 outer_io.yMin(),
+                                 outer_io.xMax() - cwidth,
+                                 outer_io.yMax() - cheight);
+
+  const odb::dbTransform xform(rotation);
+
+  // Create corners
+  const int corner_sites
+      = std::max(horizontal_site->getHeight(), corner_site->getWidth())
+        / corner_site->getWidth();
+  auto create_corner = [block, corner_site, corner_sites, &name_format, &xform](
+                           const std::string& name,
+                           const odb::Point& origin,
+                           odb::dbOrientType orient) -> odb::dbRow* {
+    const std::string row_name = fmt::format(name_format, name);
+    odb::dbTransform rotation(orient);
+    rotation.concat(xform);
+    return odb::dbRow::create(block,
+                              row_name.c_str(),
+                              corner_site,
+                              origin.x(),
+                              origin.y(),
+                              rotation.getOrient(),
+                              odb::dbRowDir::HORIZONTAL,
+                              corner_sites,
+                              corner_site->getWidth());
+  };
+  auto* nw = create_corner(
+      "CORNER_NORTH_WEST", corner_origins.ul(), odb::dbOrientType::MX);
+  create_corner(
+      "CORNER_NORTH_EAST", corner_origins.ur(), odb::dbOrientType::R180);
+  auto* se = create_corner(
+      "CORNER_SOUTH_EAST", corner_origins.lr(), odb::dbOrientType::MY);
+  auto* sw = create_corner(
+      "CORNER_SOUTH_WEST", corner_origins.ll(), odb::dbOrientType::R0);
+
+  // Create rows
+  auto create_row = [block, &name_format, &xform](const std::string& name,
+                                                  odb::dbSite* site,
+                                                  int sites,
+                                                  const odb::Point& origin,
+                                                  odb::dbOrientType orient,
+                                                  odb::dbRowDir direction) {
+    const std::string row_name = fmt::format(name_format, name);
+    odb::dbTransform rotation(orient);
+    rotation.concat(xform);
+    odb::dbRow::create(block,
+                       row_name.c_str(),
+                       site,
+                       origin.x(),
+                       origin.y(),
+                       rotation.getOrient(),
+                       direction,
+                       sites,
+                       site->getWidth());
+  };
+  create_row("NORTH",
+             vertical_site,
+             x_sites,
+             {nw->getBBox().xMax(),
+              outer_io.yMax() - static_cast<int>(vertical_site->getHeight())},
+             odb::dbOrientType::MX,
+             odb::dbRowDir::HORIZONTAL);
+  create_row("EAST",
+             horizontal_site,
+             y_sites,
+             {outer_io.xMax() - static_cast<int>(horizontal_site->getHeight()),
```

Comment:
- getWidth() ?

Reply:
for the horizontal sites, the height of the site is how deep the row will be.